FIP-0093: Set the Mining Reserve to Zero (#1039)
filecoin-project/FIPs: FIPS/fip-0093.md

---
fip: "0093"
title: Set the Mining Reserve to Zero
author: jnthnvctr (@jnthnvctr)
discussions-to: https://github.com/filecoin-project/FIPs/discussions/1030 
status: Draft
type: Technical
category: Core
created: 2024-07-25
---

# FIP-0093: Set the Mining Reserve to Zero 

## Simple Summary

This proposal aims to remove the mining reserve tokens. This does not preclude other FIPs from being proposed to create new tokens (as future incentives, funding iniatives etc). After this FIP, the total supply of FIL should be ~1717066618.96 (2b minus the current amt in [f090](https://filfox.info/en/address/f090)) - though future FIPs may change this number.

> ## Abstract 

Reviewer: ```suggestion

## Abstract 
```

This proposal seeks to remove the mining reserve by setting `f090` to have a balance of 0. This does not preclude new tokens from being created - rather that any new incentives should be created at the point when new forms of mining are introduced rather than relying on a set of pre-minted tokens.

## Change Motivation

There are a number of reasons why the mining reserve can be viewed as a net cost: 
1. The role of the mining reserve is non-standard - at the time of authoring, no other L1 has this amount of supply left up to potential release (~50% of the current supply, 15% of the total max supply) which creates a large gap between the FDV and Market Cap.
2. The mining reserve requires a FIP to use (the same process for creating new tokens) - in the event these tokens are needed; its unclear there would be any _less_ pushback in the community if were to create new sources of dilution.

> ## Spec

Reviewer: ```suggestion

## Spec
```

---
fip: "0093"
title: Set the Mining Reserve to Zero
author: jnthnvctr (@jnthnvctr)
discussions-to: https://github.com/filecoin-project/FIPs/discussions/1030 
status: Draft
type: Technical
category: Core
created: 2024-07-25
---

> 
> 
> 
> # FIP-0093: Set the Mining Reserve to Zero 
> 
> 

Reviewer 2: ```suggestion

# FIP-0093: Set the Mining Reserve to Zero 

```

## Simple Summary

This proposal aims to remove the mining reserve tokens. This does not preclude other FIPs from being proposed to create new tokens (as future incentives, funding iniatives etc). After this FIP, the total supply of FIL should be ~1717066618.96 (2b minus the current amt in [f090](https://filfox.info/en/address/f090)) - though future FIPs may change this number.

> 
> 

Reviewer 2: ```suggestion

```

This proposal seeks to remove the mining reserve by setting `f090` to have a balance of 0. This does not preclude new tokens from being created - rather that any new incentives should be created at the point when new forms of mining are introduced rather than relying on a set of pre-minted tokens.

## Change Motivation

There are a number of reasons why the mining reserve can be viewed as a net cost: 
1. The role of the mining reserve is non-standard - at the time of authoring, no other L1 has this amount of supply left up to potential release (~50% of the current supply, 15% of the total max supply) which creates a large gap between the FDV and Market Cap.
2. The mining reserve requires a FIP to use (the same process for creating new tokens) - in the event these tokens are needed; its unclear there would be any _less_ pushback in the community if were to create new sources of dilution.

> 
> 
> 

Reviewer 2: ```suggestion

```

## Spec

1. Set balance of `f090` to 0.

> 2. Adjust all circulating supply dependencies and the invariant checks to recognize the total supply is no longer 2b given the mining reserve has been set to 0. 

Reviewer 2: We have a `TotalFilecoin` that's used for a few things:

1. Invariant check as mentioned here, after summing up all the accounts it checks that it's equal to this number: https://github.com/filecoin-project/go-state-types/blob/aff1b8dbe7fd175eb2d9e7ca7e5da00e125a5650/builtin/v15/check.go#L203-L206
2. 3 exported methods: `DealClientCollateralBounds`, `DealPricePerEpochBounds` and `DealProviderCollateralBounds` - each of which return a min and a max, where the max is this `TotalFilecoin`. In Lotus at least, we no longer use the first two, but the last is exposed as `StateDealProviderCollateralBounds`, which is being codified in the Common Node API @ https://github.com/filecoin-project/FIPs/pull/1027

Should we find out precisely what this new number should be and replace it with that? Or do something else with these APIs?

Reviewer 2: The new total supply would be exactly `1717066618961773405230063046` attoFIL. That's current `TotalFilecoin` minus f090's balance.

So we could do this?

```suggestion
2. Adjust all circulating supply dependencies and the invariant checks to recognize the total supply is no longer 2b given the mining reserve has been set to 0. The new total supply value used for invariant checks and other uses of "maximum Filecoin", should be set to `1717066618961773405230063046` attoFIL.
```

Reviewer 3: I believe total supply is defined to include the burned balance. It's the 2bn sum of all address balances.

Reviewer 2: The invariant check does a total of all address balances to get to 2b. If we're not burning, but setting to `0` as per this FIP then we're expecting a new total of `1717066618961773405230063046`.

## Security Considerations

This should not affect the security as these tokens are not used to incentivize any action. Any future proposal that might need new sources of inflation to help incentivize / secure new mechanisms are not blocked from including new incentives in their proposals.

## Incentive Considerations

The proposed change requires future explicit proposals (aiming to incentivize new forms of mining) to explicitly mint new tokens. This should not change the difficulty of making such proposals, as both (using the mining reserve, creating new tokens) would have the short term effect of diluting stakeholders. This changes the optics of having large amounts of potential dilution that do not have a definite time horizon to entering the supply (which can have a negative signal).

## Implementations

1. Set the balance of `f090` to 0.

> 2. Adjust all circulating supply dependencies and the invariant checks to recognize the total supply is no longer 2b given the mining reserve has been set to 0. The total supply of the token should be the total supply of FIL should be ~1717066618.96 (2b minus the current amt in [f090](https://filfox.info/en/address/f090)).

Reviewer 2: ```suggestion
2. Adjust all circulating supply dependencies and the invariant checks to recognize the total supply is no longer 2b given the mining reserve has been set to 0. The total supply of the token should be the total supply of FIL should be `1717066618961773405230063046` attoFIL (2b minus the current amt in [f090](https://filfox.info/en/address/f090)).
```

> 
> 

Reviewer 2: ```suggestion

```
